OptimobileShared/Network/NetworkClient.swift

```diff
@@ -53,14 +61,8 @@ class KSHttpClient {
         authHeader = nil
     }
 
-    func setBasicAuth(user: String, password: String) {
-        let creds = "\(user):\(password)"
-        let data = creds.data(using: .utf8)
-        let base64Creds = data?.base64EncodedString()
-
-        if let encoded = base64Creds {
-            authHeader = "Basic \(encoded)"
-        }
+    deinit {
+        invalidateSessionCancellingTasks(true)
```

Comment:
AnalyticsHelper has `invalidateSessionCancellingTasks(false)`. 

now every HttpClient does this. Does it mean invalidateSessionCancellingTasks is going to be called 2 times with different values?

Reply:
Good catch. I didn't notice it. Move this function with the correct argument to a holding object